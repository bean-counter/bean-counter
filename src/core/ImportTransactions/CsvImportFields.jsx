/* eslint no-console: 0 */
import React from 'react'
import PropTypes from 'prop-types'
import TextField from '@material-ui/core/TextField'

const CsvImportFields = ({ institution, values, handleChange }) => {
  switch (institution) {
    case 'BMO':
      return (
        <div>
          <TextField
            label="Currency"
            inputProps={{
              'aria-label': 'Currency',
              required: true,
              maxLength: 6
            }}
            value={values.ticker}
            name="ticker"
            onChange={handleChange}
          />
        </div>
      )
    case 'TD':
    case 'Tangerine':
      return (
        <div>
          <TextField
            label="Account"
            inputProps={{
              'aria-label': 'Account',
              required: true,
              maxLength: 100
            }}
            value={values.account}
            name="account"
            helperText="The name of the account (e.g. RRSP, TFSA, etc)"
            onChange={handleChange}
          />
          <TextField
            label="Currency"
            inputProps={{
              'aria-label': 'Currency',
              required: true,
              maxLength: 6
            }}
            value={values.ticker}
            name="ticker"
            onChange={handleChange}
          />
        </div>
      )
    default:
      return null
  }
}

CsvImportFields.propTypes = {
  institution: PropTypes.string.isRequired
}

export default CsvImportFields
