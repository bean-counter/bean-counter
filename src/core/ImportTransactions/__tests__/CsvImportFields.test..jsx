import React from 'react'
import renderer from 'react-test-renderer'
import CsvImportFields from '../CsvImportFields'

describe('CsvImportFields', () => {
  const mockHandleChange = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot for BMO', () => {
    const component = renderer.create((
      <CsvImportFields
        handleChange={mockHandleChange}
        institution="BMO"
        values={{ ticker: 'CAD' }}
        classes={{ }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot for TD', () => {
    const component = renderer.create((
      <CsvImportFields
        handleChange={mockHandleChange}
        institution="TD"
        values={{ ticker: 'CAD' }}
        classes={{ }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot for Tangerine', () => {
    const component = renderer.create((
      <CsvImportFields
        handleChange={mockHandleChange}
        institution="Tangerine"
        values={{ ticker: 'CAD' }}
        classes={{ }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
