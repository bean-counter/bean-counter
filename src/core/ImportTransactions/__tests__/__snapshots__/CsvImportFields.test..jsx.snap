// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsvImportFields matches snapshot for BMO 1`] = `
<div
  className="CsvImportFields-root-1"
>
  <p
    className="MuiTypography-root-3 MuiTypography-body1-12 MuiTypography-gutterBottom-30"
  >
    Fill out the following field
  </p>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-53 MuiFormLabel-filled-57 MuiInputLabel-root-42 MuiInputLabel-formControl-47 MuiInputLabel-animated-50 MuiInputLabel-shrink-49"
      data-shrink={true}
    >
      Currency
    </label>
    <div
      className="MuiInputBase-root-73 MuiInput-root-60 MuiInput-underline-64 MuiInputBase-formControl-74 MuiInput-formControl-61"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Currency"
        className="MuiInputBase-input-83 MuiInput-input-68"
        disabled={false}
        maxLength={6}
        name="ticker"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value="CAD"
      />
    </div>
  </div>
</div>
`;

exports[`CsvImportFields matches snapshot for TD 1`] = `
<div
  className="CsvImportFields-root-1"
>
  <p
    className="MuiTypography-root-3 MuiTypography-body1-12 MuiTypography-gutterBottom-30"
  >
    A few things are missing from the CSV file
  </p>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-53 MuiInputLabel-root-42 MuiInputLabel-formControl-47 MuiInputLabel-animated-50"
      data-shrink={false}
    >
      Account
    </label>
    <div
      className="MuiInputBase-root-73 MuiInput-root-60 MuiInput-underline-64 MuiInputBase-formControl-74 MuiInput-formControl-61"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Account"
        className="MuiInputBase-input-83 MuiInput-input-68"
        disabled={false}
        maxLength={100}
        name="account"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
    <p
      className="MuiFormHelperText-root-90"
    >
      The name of the account (e.g. Checking, Savings, RRSP, etc)
    </p>
  </div>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-53 MuiFormLabel-filled-57 MuiInputLabel-root-42 MuiInputLabel-formControl-47 MuiInputLabel-animated-50 MuiInputLabel-shrink-49"
      data-shrink={true}
    >
      Currency
    </label>
    <div
      className="MuiInputBase-root-73 MuiInput-root-60 MuiInput-underline-64 MuiInputBase-formControl-74 MuiInput-formControl-61"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Currency"
        className="MuiInputBase-input-83 MuiInput-input-68"
        disabled={false}
        maxLength={6}
        name="ticker"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value="CAD"
      />
    </div>
  </div>
</div>
`;

exports[`CsvImportFields matches snapshot for Tangerine 1`] = `
<div
  className="CsvImportFields-root-1"
>
  <p
    className="MuiTypography-root-3 MuiTypography-body1-12 MuiTypography-gutterBottom-30"
  >
    A few things are missing from the CSV file
  </p>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-53 MuiInputLabel-root-42 MuiInputLabel-formControl-47 MuiInputLabel-animated-50"
      data-shrink={false}
    >
      Account
    </label>
    <div
      className="MuiInputBase-root-73 MuiInput-root-60 MuiInput-underline-64 MuiInputBase-formControl-74 MuiInput-formControl-61"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Account"
        className="MuiInputBase-input-83 MuiInput-input-68"
        disabled={false}
        maxLength={100}
        name="account"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
    <p
      className="MuiFormHelperText-root-90"
    >
      The name of the account (e.g. Checking, Savings, RRSP, etc)
    </p>
  </div>
  <div
    className="MuiFormControl-root-38 CsvImportFields-textField-2"
  >
    <label
      className="MuiFormLabel-root-53 MuiFormLabel-filled-57 MuiInputLabel-root-42 MuiInputLabel-formControl-47 MuiInputLabel-animated-50 MuiInputLabel-shrink-49"
      data-shrink={true}
    >
      Currency
    </label>
    <div
      className="MuiInputBase-root-73 MuiInput-root-60 MuiInput-underline-64 MuiInputBase-formControl-74 MuiInput-formControl-61"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        aria-label="Currency"
        className="MuiInputBase-input-83 MuiInput-input-68"
        disabled={false}
        maxLength={6}
        name="ticker"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value="CAD"
      />
    </div>
  </div>
</div>
`;
